platform/src/main/java/org/stellar/anchor/platform/configurator/ConfigReader.java

```diff
@@ -50,6 +54,27 @@ public void validate(ConfigMap configMap) throws InvalidConfigException {
     }
   }
 
+  /**
+   * If the name is an array field, return the name without the array index and truncate the rest of
+   * the name.
+   *
+   * <p>Example: "assets[0].code" -> "assets"
+   *
+   * @param fieldName
+   * @return name of the array
+   */
+  String trimToArrayName(String fieldName) {
+    String regex = "\\[\\d+\\]";
+    Pattern pattern = Pattern.compile(regex);
```

Comment:
Nit: if we compile this every time we match, we may as well not compile it.

Reply:
Optimized.